flaml/automl.py

```diff
@@ -1502,6 +1607,14 @@ def custom_metric(
         self._auto_augment = auto_augment
         self._min_sample_size = min_sample_size
         self._prepare_data(eval_method, split_ratio, n_splits)
+
+        if _is_nlp_task(self._state.task):
+            if X_val is not None:
+                self._state.fit_kwargs["X_val"] = X_val
+            if y_val is not None:
+                self._state.fit_kwargs["y_val"] = y_val
```

Comment:
move to _compute_with_config_base(); don't modify fit_kwargs in place

Reply:
moved;